testslide/lib.py

```diff
@@ -60,11 +60,18 @@ def _validate_callable_signature(
 
 
 def _validate_argument_type(expected_type, name: str, value) -> None:
-    if _is_a_mock(value):
-        template = _extract_mock_template(value)
-        if not template:
-            return
-    typeguard.check_type(name, value, expected_type)
+    original_check_type = typeguard.check_type
+
+    def wrapped_check_type(argname, inner_value, inner_expected_type, *args, **kwargs):
+        if _is_a_mock(inner_value):
+            if _extract_mock_template(inner_value) is None:
+                return
+        return original_check_type(
+            argname, inner_value, inner_expected_type, *args, **kwargs
+        )
+
+    with unittest.mock.patch.object(typeguard, "check_type", new=wrapped_check_type):
+        typeguard.check_type(name, value, expected_type)
```

Comment:
Just note that any other module that did an import of typeguard, will see the patched version for the little time that this with runs, so it might lead to very weird errors if running in parallel, maybe we should monkey-patch at the beginning only, as un-patching might also interfere, wdyt?

Reply:
I agree, but not sure how to fix. I put the patch with the smallest scope possible (might even take performance away...) precisely because of that... Not sure which's worse.  Hopefully typeguard will get this improved upstream, and we'll get rid of this soon.
Feel free to cut a PR to change this if you feel passionate, I'll gladly merge it.